presto-clp/src/main/java/com/facebook/presto/plugin/clp/metadata/filter/ClpMetadataFilterProvider.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.plugin.clp.metadata.filter;
+
+import com.facebook.presto.plugin.clp.ClpConfig;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.SchemaTableName;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+
+import static com.facebook.presto.plugin.clp.ClpConnectorFactory.CONNECTOR_NAME;
+import static com.facebook.presto.plugin.clp.ClpErrorCode.CLP_MANDATORY_METADATA_FILTER_NOT_VALID;
+import static com.facebook.presto.plugin.clp.ClpErrorCode.CLP_METADATA_FILTER_CONFIG_NOT_FOUND;
+import static com.facebook.presto.plugin.clp.metadata.filter.ClpMetadataFilter.MetadataDatabaseSpecific;
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Loads and manages metadata filter configurations for the CLP connector.
+ * <p></p>
+ * The configuration file is specified by the {@code clp.metadata-filter-config} property
+ * and defines metadata filters used to optimize query execution through split pruning.
+ * <p></p>
+ * Filter configs can be declared at either a catalog, schema, or table scope. Filter configs under
+ * a particular scope will apply to all child scopes (e.g., schema-level filter configs will apply
+ * to all tables within that schema).
+ * <p></p>
+ * For different type of metadata database, each filter config could include different following
+ * fields. See {@link ClpMetadataFilter} for the structure definition.
```

Comment:
```suggestion
 * For different metadata databases filter configs can include different fields.
 * See {@link ClpMetadataFilter} for the structure definition.
```
Not quite sure if my understanding of "different following fields" is correct/not sure if "can include different fields" is specific enough.

My interpretation is that we're trying to convey that there can be customization for different metadata databases inside of the `metadataDatabaseSpecific` field. Not sure if what I rewrote here is correct/clear enough or if we want to ad some more detail though.